### PR TITLE
Fix mutable binding creation for realms with proxy this binding

### PR DESCRIPTION
**Pull request by zherczeg**

### CODE REVIEW

**Review by rerobika — APPROVED** [review]
LGTM

File: jerry-core/ecma/operations/ecma-lex-env.c [hunk 1]
```diff
@@ -175,10 +176,31 @@ ecma_op_create_mutable_binding (ecma_object_t *lex_env_p, /**< lexical environme
...
     ecma_object_t *binding_obj_p = ecma_get_lex_env_binding_object (lex_env_p);
 
+#if ENABLED (JERRY_BUILTIN_PROXY) && ENABLED (JERRY_BUILTIN_REALMS)
+    if (ECMA_OBJECT_IS_PROXY (binding_obj_p))
+    {
+      ecma_value_t result = ecma_proxy_object_is_extensible (binding_obj_p);
+
+      if (ECMA_IS_VALUE_ERROR (result))
+      {
+        return result;
```
Comment (galpeter) [inline jerry-core/ecma/operations/ecma-lex-env.c:186]: Could we add a test for this case?

**Comment by zherczeg** [comment]
Thank you for the review. Patch is updated.

**Review by galpeter — APPROVED** [review]
lgtm

**Pull request merged**